doc/tutorials/ostro-oic-tutorial.md

```diff
@@ -0,0 +1,1030 @@
+```
+
+If you run the application now, you'll see that it now keeps 'state' and
+'power' values between executions. As we are using 'filesystem'
+persistence, you may notice two new files on the dir you are running
+application: 'state' and 'power', exact the same name given to
+persistence nodes on our subflow.
+
+## Step 4
+
+### Controlling a remote light using OIC
+
+So far, we stayed on only one one device. But what if we want to, say,
+control a remote light? Soletta also supports [OIC](http://openinterconnect.org/)
+and has a 'light client/server' sample. We are going to use the
+controller we built so far to control this remote light.
+We are going to use our controller, but let's modify it to be a subflow
+as well. So we can use it as a generic controller.
+We start by exposing its relevant ports:
+
+```
+OUTPORT=power_value.OUT:POWER
+OUTPORT=state_value.OUT:STATE
+
+INPORT=power_value.SET:POWER
+INPORT=state_value_from_boolean.IN:STATE
+```
+
+Now, our controller subflow will simply have `POWER` and `STATE`
+ports in both directions (`IN` and `OUT`).
+**Note:** As it doesn't make sense for a client controller store data, we
+should also remove persistence related nodes. We will move persistence
+stuff to our fbp servers later on.
+**Note:** It's also a good idea to change controller lines that output
+status to something more meaningful, like:
+print content,
+
+```
+power_value OUT -> IN power_value_output(console:prefix="Power value changed: ")
+state_value OUT -> IN state_value_output(console:prefix="State changed: ")
+```
+
+so they are easily identified on output.
+
+On a new file, `oic-controller.fbp` for instance, declare our controller
+subflow (and create it):
+
+```
+DECLARE=controller:fbp:controller.fbp
+
+# create a controller
+controller(controller)
+```
+
+OIC resources, like Soletta light-server sample, have an ID to uniquely
+identify them. As our client is going to connect to a specific resource,
+we need to define its ID somehow. We are going to get them from command
+line argument used to run our application. This can be done with help
+of node `app/argc-argv`:
+
+```
+cmdline_arguments(app/argc-argv)
+_(constant/int:value=1) OUT -> IN cmdline_arguments
+```
+
+With this, when we get output value from `cmdline_arguments` node, it
+will output command line argument number 1.
+Then, we use Soletta `oic/client-light` [5] to communicate with a light
+server. We also cross link it our controller, so changes on one are
+properly reflected on another - note however, that at this point, changes
+only happen in the direction 'controller -> oic/client', but we'll
+improve this later.
+
+```
+oic(oic/client-light)
+
+# Uses command line argument as device id
+cmdline_arguments OUT -> DEVICE_ID oic
+
+# Cross link OIC and controller
+oic OUT_STATE -> STATE controller
+oic OUT_POWER -> POWER controller
+
+controller STATE -> IN_STATE oic
+controller POWER -> IN_POWER oic
+```
```

Comment:
explain the cross-link properties here, I've found that people don't get that it will be stable and not loop infinitely.